# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane 4: CTR / Engagement Opportunity Scoring

Justification: High-impression pages with low CTR or engagement represent low-hanging fruit where small tweaks (like meta titles, descriptions, or page layout) can yield immediate traffic and conversion gains without waiting for ranking updates.

In [20]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Core Research Question: Which high-volume pages are underperforming in click-through rate or user engagement compared to their expected benchmark?

Decision: Prioritizing which pages receive immediate content/metadata optimizations versus full content rewrites.

Who Acts: Content strategists, SEO specialists, and copywriters.

Recommended Action: Update meta titles/descriptions to boost search CTR, or improve above-the-fold content to increase engagement and scroll depth.

Cost of a Wrong Call: Wasted editorial hours rewriting non-essential content or accidentally altering metadata on high-performing pages, leading to a drop in organic traffic.

In [21]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [22]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load directly from your GitHub repository (replace 'Pauloade123' if using a different handle)
url = "https://raw.githubusercontent.com/Pauloade123/ML-intern/main/data/raw/content_refresh_anonymized.csv"

import pandas as pd

# 1. Load the dataset from your personal repository
url = "https://raw.githubusercontent.com/Pauloade123/ML-intern/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)
df.columns

# 2. Compute the 3 supporting metrics
total_pages = len(df)
avg_ctr = df['ctr'].mean() * 100
opportunity_pages = len(df[(df['impressions_90d'] > df['impressions_90d'].median()) & (df['ctr'] < df['ctr'].mean())])

# 3. Print the results
print(f"Total pages: {total_pages:,}")
print(f"Average CTR: {avg_ctr:.2f}%")
print(f"High-impression / Low-CTR opportunity pages: {opportunity_pages:,}")



Total pages: 30,000
Average CTR: 51.07%
High-impression / Low-CTR opportunity pages: 12,866


Across a total dataset of 30,000 pages, the overall average CTR is 51.07%. However, 12,866 pages have above-median 90-day impressions while falling below the average CTR, representing a massive pool of high-visibility pages that can yield quick traffic gains from title and metadata optimizations

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I CAN claim: "We can identify observed pattern relationships, rank candidate pages by relative opportunity scores, and provide decision-support priorities for editorial teams

What I CANNOT claim: "We cannot claim direct causal proof, guarantee ranking improvements, or 'predict Google's algorithm,' as search position depends on external market forces and unobserved competitor changes.

In [23]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.